firmware/libraries/system_time/system_time.c

```diff
@@ -40,22 +53,33 @@ inline uint32_t system_time_cmp_us(uint32_t old_time, uint32_t new_time)
 {
   uint32_t tick_diff = system_time_cmp_ticks(old_time, new_time);
 
-  return TICKS_TO_US(tick_diff);
+  return ST_TICKS_TO_US(tick_diff);
 }
 
 inline uint32_t system_time_cmp_ms(uint32_t old_time, uint32_t new_time)
 {
   uint32_t tick_diff = system_time_cmp_ticks(old_time, new_time);
 
-  return TICKS_TO_MS(tick_diff);
+  return ST_TICKS_TO_MS(tick_diff);
 }
 
 uint32_t system_time_get(void)
 {
-  return SYSTEM_TIME_TIMER->CNT;
+  DISABLE_IRQ();
+  uint32_t time = (AUTORELOAD * s.overrun_cnt) + SYSTEM_TIME_TIMER->CNT;
+  ENABLE_IRQ();
+  return time;
 }
 
 void system_time_init(void)
 {
   timer_init();
 }
+
+void SYSTEM_TIME_TIMER_IRQHandler(void)
+{
+  DISABLE_IRQ();
+  LL_TIM_ClearFlag_UPDATE(SYSTEM_TIME_TIMER);
+  s.overrun_cnt++;
+  ENABLE_IRQ();
+}
```

Comment:
I think there is potentially a conflict in the error handler functionality in main now.

The error handler disables irq, but also tries to call delay_ms which is dependant on irq

Reply:
good catch